src/parser/expression.js

```diff
@@ -588,12 +592,22 @@ export default class ExpressionParser extends LValParser {
     node.property = this.parseIdentifier(true);
 
     if (node.property.name !== propertyName) {
-      this.raise(node.property.start, `The only valid meta property for new is ${meta.name}.${propertyName}`);
+      this.raise(node.property.start, `The only valid meta property for ${meta.name} is ${meta.name}.${propertyName}`);
     }
 
     return this.finishNode(node, "MetaProperty");
   }
 
+  parseImportMetaProperty(): N.MetaProperty {
+    const node = this.startNode();
+    const id = this.parseIdentifier(true);
+    this.expect(tt.dot);
+    if (this.options.sourceType !== "module") {
```

Comment:
Weird, I thought I tried that first but `.inModule` was only true top-level. Must have missed something. You're right, definitely works fine.